staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -149,7 +149,8 @@ func (a *AdmissionOptions) ApplyTo(
 		return err
 	}
 
-	c.AdmissionControl = admissionmetrics.WithStepMetrics(admissionChain)
+	admissionControl := withReinvocation(admissionChain)
```

Comment:
this looks like the wrong place for the reinvocation logic. All this is deeply connected with the admission chain. Hence, I would rather expect this in k8s.io/apiserver/pkg/admission.

Reply:
Makes sense. I've moved this into the code that instantiates and provides the admission chain to the server.